https://bugs.gentoo.org/851702#c4 Adapt to libLLVM.so instead of the seperated
libs. Compare to LLVM-15, do not re-link libLLVMSupport to fix 
"CommandLine Error: Option 'debug-counter' registered more than once!"
===================================================================
Index: ROCm-Device-Libs-rocm-5.4.2/utils/prepare-builtins/CMakeLists.txt
===================================================================
--- ROCm-Device-Libs-rocm-5.4.2.orig/utils/prepare-builtins/CMakeLists.txt
+++ ROCm-Device-Libs-rocm-5.4.2/utils/prepare-builtins/CMakeLists.txt
@@ -26,5 +26,4 @@ set_target_properties(prepare-builtins P
   CXX_STANDARD_REQUIRED Yes
   CXX_EXTENSIONS No)
 llvm_update_compile_flags(prepare-builtins)
-llvm_map_components_to_libnames(llvm_libs support core bitreader bitwriter)
-target_link_libraries(prepare-builtins ${llvm_libs})
+target_link_libraries(prepare-builtins LLVM)
